chef_master/source/server_security.rst

```diff
@@ -268,6 +268,55 @@ To regenerate SSL certificates:
 
       $ chef-server-ctl start
 
+Chef Server Credentials Management
+=====================================================
+
+Since 12.14.0
+-----------------------------------------------------
+
+Chef server limits where it writes service passwords and keys to disk. In the default configuration, credentials are only written to files in ``/etc/opscode``.
+
+By default, to maintain compatibility with older add-on versions, Chef server writes service credentials to multiple locations inside ``/etc/opscode``. However, if you are not using add-ons or are using the latest releases of your add-ons, a new configuration item ``insecure_addon_compat`` allows you to further restrict where credentials are written. When set to ``false``, credentials are only written to ``/etc/opscode/private-chef-secrets.json``.
+
+User-provided secrets (such as the password for an external postgresql instance) can still be set in ``/etc/opscode/chef-server.rb`` or via the :ref:`ctl_chef_server_secrets_management` commands.  These commands allow you to provide external passwords without including them in your configuration file.
+
+Add-on Compatibility
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+The following table lists which add-on versions support the more restrictive ``insecure_addon_compat false`` setting. These version also now **require** Chef server 12.14.0 or greater:
+
+.. list-table::
+   :widths: 1 1
+   :header-rows: 1
+
+   * - Add-on Name
+     - Minimum Version
+   * - Chef Backend
+     - *all*
+   * - Chef Manage
+     - 2.5.0
+   * - Push Jobs Server
+     - 2.2.0
+   * - Reporting
+     - 1.7.0
+   * - Analytics
+     - *none*
+
+These newer add-ons will also write all of their secrets to ``/etc/opscode/private-chef-secrets.json``. Older versions of the add-ons will still write their configuration to locations in ``/etc`` and ``/var/opt``.
+
+/etc/opscode/private-chef-secrets.json
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+``/etc/opscode/private-chef-secrets.json``'s default permissions only allow the root user to read or write the file. This file contains all of the secrets for access to the Chef server's underlying data stores, and thus access to it should be restricted to trusted users.
+
+While the file does not contain passwords in plaintext, it is not safe to share with untrusted users. The format of the secrets file, allows deployments of Chef server to conform to regulations requiring that no sensitive data should appear in plain text in configuration files; however, it does not make the file meaningfully more secure.
```

Comment:
I'v fixed the first comma, but I feel like the comma after regulations would change the meaning here.